### PR TITLE
chore(e2e): Add support for variable binary names for end-to-end tests

### PR DESCRIPTION
**Pull request by joshbrand**
This PR updates enos workflows to add support for variable binary names in order to test various Boundary builds.

### CODE REVIEW

File: enos/enos-variables.hcl [hunk 1]
```diff
@@ -154,3 +154,15 @@ variable "docker_mirror" {
   type        = string
   default     = "docker.mirror.hashicorp.services"
 }
+
+variable "boundary_license" {
```
Comment (moduli) [inline enos/enos-variables.hcl:158]: I assume this is a file path to a license file? Could we specify that somewhere so it's specific (whether it's the description of the variable name)

File: enos/enos-modules.hcl [hunk 2]
```diff
@@ -22,8 +22,10 @@ module "boundary" {
     "Environment" : var.environment
   }
 
-  ssh_aws_keypair       = var.aws_ssh_keypair_name
```
Comment (moduli) [inline enos/enos-modules.hcl:25]: Was this moved down to be sorted alphabetically? If so, then `boundary_binary_name` and `boundary_license` should be swapped?

**Review by moduli — COMMENTED** [review]
Some minor comments

File: enos/enos-modules.hcl (hunk 2 above)
Comment (joshbrand) [reply enos/enos-modules.hcl:25]: spelling is hard!

File: enos/enos-variables.hcl (hunk 1 above)
Comment (joshbrand) [reply enos/enos-variables.hcl:158]: Actually it's the license string, I'll put in code to read it from a file for ease though!

**Comment by moduli** [comment]
Closing for now. Rethinking approach.
